test/descriptive-link-text.md

```diff
@@ -0,0 +1,23 @@
+# Descriptive link text
+
+[Learn about Javascript](https://example.com/javascript/about)
+
+[Read about Javascript](https://example.com/javascript/about)
+
+Learn about [JavaScript](https://example.com/javascript/about).
+
+[Learn more](https://example.com/javascript/about). {MD059}
+
+[Click here](https://example.com/javascript/about). {MD059}
+
+To learn more, go [here!](https://example.com/javascript/about). {MD059}
+
+To learn more, [click here!!!!](https://example.com/javascript/about). {MD059}
+
+[click-here!!!!](https://example.com/javascript/about). {MD059}
+
+Go to this [link]((https://example.com/javascript/about)). {MD059}
+
+[link][1] {MD059}
```

Comment:
@DavidAnson Is this what you meant by reference link? 
Do you have an example of a "shortcut link"?

related: https://github.com/DavidAnson/markdownlint/pull/1459#discussion_r1904887361

Reply:
Examples here: https://github.com/DavidAnson/markdownlint/blob/main/doc/md054.md

Reply:
Yours is a reference link, but it looks more like a footnote. Let's use something more typical as shown in the examples I link.